python/tank/platform/engine.py

```diff
@@ -1774,14 +1803,26 @@ def _define_qt_base(self):
                 base["dialog_base"] = importer.QtGui.QDialog
             else:
                 base["dialog_base"] = None
-            base["wrapper"] = importer.wrapper
         except:
 
             self.log_exception("Default engine QT definition failed to find QT. "
                                "This may need to be subclassed.")
 
         return base
 
+    def __define_qt5_base(self):
+        """
+        This will be called at initialization to discover every PySide 2 modules. It should provide
+        every Qt modules available as well as two extra attributes, ``__name__`` and
+        ``__version__``, which refer to the name of the binding and it's version, e.g.
+        PySide2 and 2.0.1.
+
+        .. note:: PyQt5 not supported since it runs only on Python 3.
+
+        :returns: A dictionary with all the modules, __version__ and __name__.
+        """
+        return QtImporter(interface_version_requested=QtImporter.Qt5).base
```

Comment:
If i understand the code correctly, this will actually attempt to import the qt5 payload, resulting in lots of file I/O, loading of dlls etc. I may be overly catious here, but I think it would be better if the probing phase here could be a little more lightweight if possible - could we check for QT5 without doing a full import? The current implementation means that every engine that starts up will dive deep into QT, load everything up, even in cases where it's not needed. 

In the past, we had some odd cases where even a try/catch around qt imports wouldn't cut it because qt itself would `abort` and drag everything else with in a gigantic crash. I think there are still cases when QT does this - although i admit it maybe be a far fetched use case! :) 

I am also wondering about scenarios where you may want the engine to do something (like set some DLL paths or something) before the actual import happens. I think there may be a difference between "i have access to QT" and "I am importing QT". Does that make sense?